utils/s2n_mem.c

```diff
@@ -98,15 +161,24 @@ int s2n_realloc(struct s2n_blob *b, uint32_t size)
 
     /* blob already has space for the request */
     if (size < b->allocated) {
```

Comment:
what if size == b->allocated, shouldn't that just reuse the existing memory?

Reply:
done!